### PR TITLE
Tagged na

### PR DESCRIPTION
**Pull request by wyusuf068**
- Combine similar functions into one .R file.
- Examine tagged_na usage in derived functions.
- Add documentation for tagged_na in derived variables.

### CODE REVIEW

File: R/adl.R [hunk 1]
```diff
@@ -122,4 +121,160 @@ adl_fun <- function (ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
...
+#'  variables are used. All variables are common across CCHS cycles from 2001
+#'  to 2014:
+#' \enumerate{
+#'   \item ADL_01 - Needs help preparing meals
+#'   \item ADL_02 - Needs help getting to appointments/errands
+#'   \item ADL_03 - Needs help doing housework
+#'   \item ADL_04 - Needs help doing personal care
+#'   \item ADL_05 - Needs help moving inside house
+#' }
+#' There are two categories to represent "missing" and "not applicable".
```
Comment (DarkShadeKnigh) [inline R/adl.R:139]: I know you mention the tagged nas in  the return but this sentance feels incomplete. I suggest adding some reference to the tagged na here as well.

File: R/adl.R [hunk 2]
```diff
@@ -122,4 +121,160 @@ adl_fun <- function (ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
...
+#'   \item 3 - Needs help with at least 3 tasks
+#'   \item 4 - Needs help with at least 4 tasks
+#'   \item 5 - Needs help with at least 5 tasks
+#'   \item NA(a) - Not applicable. At least one of the component variables
+#'    had a value of not applicable
+#'   \item NA(b) - Missing. At least one of the component variables had a value
+#'    of missing, but none of had a **not applicable** value
+#' }
+#'
+#' @export
```
Comment (DarkShadeKnigh) [inline R/adl.R:161]: Could you move the export after the examples so its easier to see which functions are exported

File: R/adl.R [hunk 3]
```diff
@@ -122,4 +121,160 @@ adl_fun <- function (ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
...
+#'
+#' ADL_score_5 <- adl_score_5_fun(2, 1, 1, 2, 2)
+#'
+#' print(ADL_score_5)
+#'
+adl_score_5_fun <-
+  function(ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
+    # For each row in the ADL variables, count the total number of "missing"
+    # values in each row
+    count_missing_adl <- rep(0, length(ADL_01))
```
Comment (DarkShadeKnigh) [inline R/adl.R:210]: Why is length of ADL_01 used when its only passed a single value meaning its going to always be 1?

File: R/adl.R [hunk 4]
```diff
@@ -122,4 +121,160 @@ adl_fun <- function (ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
...
+    count_missing_adl <-
+      ifelse(ADL_03 == "NA(b)", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_04 == "NA(b)", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_05 == "NA(b)", count_missing_adl + 1, count_missing_adl)
+    
+    # For each row in the ADL variables, count the total number of "not
+    # applicable" values in each row
+    count_not_applicable_adl <- rep(0, length(ADL_01))
```
Comment (DarkShadeKnigh) [inline R/adl.R:227]: How is this different from line 110?

File: R/adl.R [hunk 5]
```diff
@@ -122,4 +121,160 @@ adl_fun <- function (ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
...
+             count_not_applicable_adl + 1,
+             count_not_applicable_adl)
+    count_not_applicable_adl <-
+      ifelse(ADL_04 == "NA(a)",
+             count_not_applicable_adl + 1,
+             count_not_applicable_adl)
+    count_not_applicable_adl <-
+      ifelse(ADL_05 == "NA(a)",
+             count_not_applicable_adl + 1,
+             count_not_applicable_adl)
```
Comment (DarkShadeKnigh) [inline R/adl.R:250]: This is too similar to your NA(b) code there should be a way to collapse it together

File: R/age-categorical.R [hunk 6]
```diff
@@ -84,7 +84,8 @@ age_cat_fun <- function(DHHGAGE_cont) {
                               (DHHGAGE_cont %in% 70:74), 14,
                               if_else2(
                                 (DHHGAGE_cont %in% 75:79), 15,
-                                if_else2((DHHGAGE_cont >= 80), 16, NA)
+                                if_else2((DHHGAGE_cont >= 80), 16,
+                                         "NA(b)")
```
Comment (DarkShadeKnigh) [inline R/age-categorical.R:88]: Your current return documentation is now inconsistent with your adl score return documentation

File: R/respiratory-condition.R [hunk 7]
```diff
@@ -315,7 +318,211 @@ resp_condition_fun3 <-
...
+#'
+#' @export
+#' 
+COPD_Emph_der_fun1 <-
+  function(DHHGAGE_cont, CCC_91E, CCC_91F) {
+    # Argument verification
+    if ((!is_equal(CCC_91E, 1) &
+         !is_equal(CCC_91E, 2)) |
+        (!is_equal(CCC_91F, 1) &
+         !is_equal(CCC_91F, 2))) {
```
Comment (DarkShadeKnigh) [inline R/respiratory-condition.R:398]: How does this handle CCC_91E or CCC_91F being NA?

Comment (DarkShadeKnigh) [reply R/respiratory-condition.R:398]: In addition consider using the %in% function ie: CCC_91E %in% 1:2 | CCC_91F %in% 1:2

File: R/respiratory-condition.R [hunk 8]
```diff
@@ -315,7 +318,211 @@ resp_condition_fun3 <-
...
+#' tail(combined_COPD)
+#' @seealso \code{\link{COPD_Emph_der_fun2}}
+#'
+#' @export
+#' 
+
+COPD_Emph_der_fun2 <-
+  function(DHHGAGE_cont, CCC_091) {
+    # Argument verification
+    if ((!is_equal(CCC_091, 1) &
```
Comment (DarkShadeKnigh) [inline R/respiratory-condition.R:496]: Same as the previous function for addressing NA values and using %in%

**Review by DarkShadeKnigh — COMMENTED** [review]
I need some additional code clarification

File: R/respiratory-condition.R (hunk 7 above)
Comment (wyusuf068) [reply R/respiratory-condition.R:398]: > How does this handle CCC_91E or CCC_91F being NA?

Line 426 examines if both CCC_91E or CCC_91F are NA(a) to set the function return to NA(a), else NA(b).

**Pull request merged**